src/main/scala/wdltool/graph/WomGraphPrint.scala

```diff
@@ -0,0 +1,130 @@
+import wdl4s.wom.graph.GraphNodePort.{InputPort, OutputPort}
+import wdl4s.wom.graph.{GraphInputNode, GraphNodePort, GraphOutputNode}
+import wdltool.graph.GraphPrint.clusterCount
+//import wdl4s.wdl.{CallOutput, Declaration, If, Scatter, WdlCall, WdlGraphNode, _}
+import wdl4s.wom.graph.{CallNode, Graph, GraphNode, ScatterNode}
+
+import scala.collection.JavaConverters._
+
+object WomGraphPrint {
+
+  final case class WorkflowDigraph(workflowName: String, digraph: NodesAndLinks)
+  final case class NodesAndLinks(nodes: Set[String], links: Set[String])
+
+  def readFile(filePath: String): String = Files.readAllLines(Paths.get(filePath)).asScala.mkString(System.lineSeparator())
+
+  def womExecutableFromWdl(filePath: String): Executable = {
+    val namespace = WdlNamespaceWithWorkflow.load(readFile(filePath), Seq(WdlNamespace.fileResolver _)).get
+    namespace.womExecutable
+  }
+
+  def womExecutableFromCwl(filePath: String): Executable = {
+    val yaml: Yaml = readFile(filePath)
+    CwlCodecs.decodeCwl(yaml) match {
+      case Right((clt: CommandLineTool, _)) =>
+        clt.womExecutable match {
+          case Valid(wom) => wom
+        }
+      case Right((wf: Workflow, m)) if m.isEmpty =>
+        wf.womExecutable(Map.empty) match {
+          case Valid(wom) => wom
+        }
+        // TODO: Don't throw away the error information! Might as well return it:
+      case other => throw new Exception(s"Can't read $filePath")
+    }
+  }
+
+  def generateWorkflowDigraph(file: String, auxFiles: Seq[String]): WorkflowDigraph = {
+
+    val womExecutable = if (file.toLowerCase().endsWith("wdl")) womExecutableFromWdl(file) else womExecutableFromCwl(file)
+    womExecutable.graph match {
+      case Valid(graph) =>
+        val digraph = listAllGraphNodes (graph)
+        WorkflowDigraph (dotSafe(womExecutable.entryPoint.name), digraph)
+      case Invalid(errors) =>
+        throw new IllegalArgumentException(errors.toList.mkString("\n", "\n", "\n"))
+    }
+  }
+
+  private val clusterCount: AtomicInteger = new AtomicInteger(0)
+
+  private def listAllGraphNodes(graph: Graph): NodesAndLinks = {
+
+    graph.nodes foldMap nodesAndLinks
+  }
+
+  private def nodesAndLinks(graphNode: GraphNode): NodesAndLinks = graphNode match {
+    case other => defaultNodesAndLinks(graphNode)
+  }
+
+  /**
+    * For most graph nodes, we draw a single box for the node, containing the name and input and output ports.
+    * For each input/output port, we include the links to upstream ports.
+    */
+  private def defaultNodesAndLinks(graphNode: GraphNode): NodesAndLinks = {
+    val clusterName = "cluster_" + clusterCount.getAndIncrement()
+    val portNodes = (graphNode.outputPorts ++ graphNode.inputPorts) map { p =>
+      s"${graphId(p)} [shape=${graphShape(p)} label=${graphName(p)}];"
+    }
+    val links = for {
+      inputPort <- graphNode.inputPorts
+      upstreamPort = inputPort.upstream
+    } yield graphId(upstreamPort)+" -> "+graphId(inputPort)
```

Comment:
I read somewhere (and kinda agree) that when you have string concatenation it makes it way more readable (and easy to see where the final spaces will end up) to do this than to spread them out, eg
```
 val a = "Hi"
 val b = "Miguel"
 val c = a+" there "+b
// vs
 val d = a + " there " + b
```

OTOH, `s""` formatting is even better yet so I'll probably just change to that